### PR TITLE
[TwigBundle] Give some love to exception pages

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | 3.2
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -
| License       | MIT
| Doc PR        | -

Before:
before

After:
after

### CODE REVIEW

File: src/Symfony/Bridge/Twig/Extension/CodeExtension.php [hunk 1]
```diff
@@ -216,6 +215,27 @@ public function formatFileFromText($text)
     }
 
     /**
+     * @internal
+     */
+    public function formatLogMessage($message, $context)
```
Comment (stof) [inline src/Symfony/Bridge/Twig/Extension/CodeExtension.php]: second argument should be typehinted as array IMO

Comment (nicolas-grekas) [inline src/Symfony/Bridge/Twig/Extension/CodeExtension.php]: done

**Comment by fabpot** [comment]
That would be 3.3 for now.

**Comment by iltar** [comment]
Are they opened by default? If so, can you add a limit of maybe 5 or 10 logs? I've had pages where there are 100+ logs and that will make it annoying to find the exception.

**Comment by nicolas-grekas** [comment]
@iltar this PR is about fixing this page, ie make it work again; not about making it better. We could (and should) work on that for 3.3. But this one is a bug fix (placeholder are not replaced right now, and CSS needs a fix in the code block excerpt).

**Comment by fabpot** [comment]
I mis-read the code as well. 👍

**Comment by iltar** [comment]
Ah my bad, I only saw new code 😆

File: src/Symfony/Bridge/Twig/Extension/CodeExtension.php [hunk 2]
```diff
@@ -151,7 +152,7 @@ public function fileExcerpt($file, $line, $srcContext = 3)
             }
 
             for ($i = max($line - $srcContext, 1), $max = min($line + $srcContext, count($content)); $i <= $max; ++$i) {
-                $lines[] = '<li'.($i == $line ? ' class="selected"' : '').'><div class="anchor" id="line'.$i.'"></div><code>'.self::fixCodeMarkup($content[$i - 1]).'</code></li>';
+                $lines[] = '<li'.($i == $line ? ' class="selected"' : '').'><a class="anchor" name="line'.$i.'"></a><code>'.self::fixCodeMarkup($content[$i - 1]).'</code></li>';
```
Comment (ro0NL) [inline src/Symfony/Bridge/Twig/Extension/CodeExtension.php]: `a[name]` is not valid HTML5... what about `<li id="x"><code></code></li>` As anchor linking will work on any ID anyway...

Comment (nicolas-grekas) [inline src/Symfony/Bridge/Twig/Extension/CodeExtension.php]: There can be only one specific id on a page, yet this is called several times.

Comment (nicolas-grekas) [inline src/Symfony/Bridge/Twig/Extension/CodeExtension.php]: If the target is not predictable, we're going to have a hard time linking to it...

Comment (ro0NL) [inline src/Symfony/Bridge/Twig/Extension/CodeExtension.php]: I see. Maybe generate a hash of `file+line`.. or something, and only add a unique prefix for the 2nd and upcoming instances (ie. the first instance is predictable).

Comment (nicolas-grekas) [inline src/Symfony/Bridge/Twig/Extension/CodeExtension.php]: a with name works, let's not create a pile of code to solve no actual issue...

Comment (ro0NL) [inline src/Symfony/Bridge/Twig/Extension/CodeExtension.php]: Imo. it should produce valid HTML5 :).. but fair enough.

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

**Pull request merged**